docs/route-tasks.md

```diff
@@ -0,0 +1,171 @@
+
+So, how do we ensure that a Worker on the correct host will execute the next Activity that depends on the video file?
+
+This is accomplished in different ways depending on the SDK.
+
+### Java
+
+Consider the [file processing sample in Java](https://github.com/temporalio/java-samples/tree/master/src/main/java/io/temporal/samples/fileprocessing). The paradigm is to create two Workers. Each will run on the same host and thus have access to the same file system, but one will listen to a global (shared) Task Queue, which it will use to start the Workflow, and the other will listen to its own host-specific (private) Task Queue. 
+
+The following code snippet is stripped down from the [FileProcessingWorker.java sample](https://github.com/temporalio/java-samples/blob/master/src/main/java/io/temporal/samples/fileprocessing/FileProcessingWorker.java) and highlights the creation of two workers for this purpose:
+
+```java
+// Initialize a WorkerFactory
+WorkerFactory factory = WorkerFactory.newInstance(client);
+// Create a Worker that is subscribed to a shared Task Queue
+final Worker workerForCommonTaskList = factory.newWorker(globalTaskQueue);
+// Register Workflow and Activities
+// Create a Worker that is subscribed to a private TaskQueue
+final Worker workerForHostSpecificTaskList = factory.newWorker(hostSpecifiTaskQueue);
+// Register activities
+```
+
+In the [Workflow](https://github.com/temporalio/java-samples/blob/master/src/main/java/io/temporal/samples/fileprocessing/FileProcessingWorkflowImpl.java#L42), the first Activity Task is executed by the Worker listening to the global Task Queue:
+
+```java
+public FileProcessingWorkflowImpl() {
+    // Create activity clients.
+    ActivityOptions ao =
+        ActivityOptions.newBuilder()
+        // Send the Activity to the shared Task Queue
+        .setTaskQueue(globalTaskQueue)
+        .build();
+    this.defaultTaskQueueStore = Workflow.newActivityStub(StoreActivities.class, ao);
+}
+```
+
+The Activities should be designed to [return an object](https://github.com/temporalio/java-samples/blob/master/src/main/java/io/temporal/samples/fileprocessing/StoreActivities.java) that includes both the Activity result and the name of the host-specific Task Queue to which the next Activity Task will be sent.
+
+```java
+public interface StoreActivities {
+
+  final class ResultAndTaskQueuePair {
+    // Name of the host-specific  Task Qeueu
+    private final String hostspecificTaskQueue;
+    // This is the Activity result (e.g. the name or location of the file)
+    private final String activityResult;
+
+    // Get and set methods ...
+  }
+
+  // Interface methods ....
+}
+```
+
+The host-specific Task Queue is then set on [subsequent Activities](https://github.com/temporalio/java-samples/blob/master/src/main/java/io/temporal/samples/fileprocessing/FileProcessingWorkflowImpl.java#L63):
```

Comment:
'"set" on subsequent activites'

I think this isn't the most accurate way of phrasing it. Activities aren't really set, the are configured when executed.